src/drivers/voxl2_io/voxl2_io.cpp

```diff
@@ -195,11 +184,16 @@ int Voxl2IO::get_version_info()
 		_packets_sent++;
 	}
 
-	/* Read response */
-	px4_usleep(10000);
-	memset(&_read_buf, 0x00, READ_BUF_SIZE);
-	res = _uart_port->uart_read(_read_buf, READ_BUF_SIZE);
+	/* Read response, wait 500ms */
 	while(read_retries){
+		hrt_abstime now = hrt_absolute_time();
+		while (hrt_elapsed_time(&now) < 500000){
```

Comment:
This is not the way to wait for 500ms! It will just sit in this loop for 500ms chewing up CPU time. Do a px4_usleep(500000) instead so that the thread goes to sleep and doesn't waste any CPU time.

Reply:
Understood! This has been updated to use px4_usleep(500000) in commit [7f7f920](https://github.com/modalai/px4-firmware/pull/50/commits/7f7f9205a93dd58d4dbbe3e5b2d6305fdfadb29c).